packages/open-next/src/build/compileConfig.ts

```diff
@@ -1,20 +1,74 @@
 import fs from "node:fs";
+import os from "node:os";
 import path from "node:path";
 
 import { buildSync } from "esbuild";
 import { OpenNextConfig } from "types/open-next.js";
 
 import logger from "../logger.js";
+import { validateConfig } from "./validateConfig.js";
 
-export function compileOpenNextConfigNode(
-  outputDir: string,
+/**
+ * Compiles the OpenNext configuration.
+ *
+ * The configuration is always compiled for Node.js and for the edge only if needed.
+ *
+ * @param baseDir Directory where to look for the configuration.
+ * @param openNextConfigPath Override the default configuration when provided. Relative to baseDir.
+ * @param nodeExternals Externals for the Node.js compilation.
+ * @return The configuration and the build directory.
+ */
+export async function compileOpenNextConfig(
```

Comment:
This is an added function.

It calls the two existing function for Node and edge if needed.

The edge test has been moved to this function.